compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java

```diff
@@ -1012,7 +1012,24 @@ public void visit(BLangBinaryExpr binaryExpr) {
             genNode(binaryExpr.lhsExpr, this.env);
             genNode(binaryExpr.rhsExpr, this.env);
             RegIndex regIndex = calcAndGetExprRegIndex(binaryExpr);
-            emit(binaryExpr.opSymbol.opcode, binaryExpr.lhsExpr.regIndex, binaryExpr.rhsExpr.regIndex, regIndex);
+            int opCode = binaryExpr.opSymbol.opcode;
+            if (opCode == InstructionCodes.INT_RANGE) {
+                RegIndex endValRegIndex = binaryExpr.rhsExpr.regIndex;
+                if (OperatorKind.HALF_OPEN_RANGE.equals(binaryExpr.opKind)) {
```

Comment:
Rather than hardcoding this logic, we can move this logic to desugar. For example. a ..< b can be desugared into a ... (b -1). Then we don't need to special case the code generation. Can you do the change?

Reply:
Moved and fixed other issues with 3954b14